torch/_inductor/codegen/cpp.py

```diff
@@ -2007,9 +2007,27 @@ def size_hint(self):
     def codegen_loops_impl(self, loop_nest, code, worksharing):
         threads = parallel_num_threads()
         assert self.call_ranges is not None
-        par_depth = self.decide_parallel_depth(
-            self.call_ranges[: loop_nest.max_parallel_depth()], threads
-        )
+
+        def get_outer_loop_fused_kernel(loop_nest):
+            # Since OuterLoopFusedKernel has no main/tail loop split at any
+            # outer loop fusion depth, len(loop_nest.root) must equal to 1.
+            if loop_nest.root and len(loop_nest.root) == 1:
+                kernels = loop_nest.root[0].get_kernels()
+                if any(isinstance(kernel, OuterLoopFusedKernel) for kernel in kernels):
+                    assert len(kernels) == 1
+                    return kernels[0]
+            return None
+
+        if kernel := get_outer_loop_fused_kernel(loop_nest):
```

Comment:
perhaps it is cleaner to just add `get_kernels` method on `LoopNestWithSplit` class?

Reply:
Thanks for the suggestion. Sure, it's clearer by adding `get_kernels` method on `LoopNestWithSplit`.